File: binary-search-iterative/binary-search-iterative.js
const numbers = [0, 2, 4, 6, 8, 10, 12, 14];

function binarySearchIterative(array, target) {
  let start = 0;
  let end = array.length - 1;
  let middle;

  while (start <= end) {
    middle = Math.round((start + end) / 2);
    if (target === array[middle]) {
      return middle;
    }

    if (target < array[middle]) {
      end = middle - 1;
    }

    if (target > array[middle]) {
      start = middle + 1;
    }
  }
  return -1;
}

console.log(binarySearchIterative(numbers, 2));

File: binary-search-iterative/binary-search-iterative.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int BinarySearchIterative(int *array, int length, int target)
{
	int start = 0;
	int end = length - 1;
	int middle;

	while (start <= end)
	{
		middle = (start + end) / 2;
		if (target == array[middle])
		{
			return middle;
		}

		if (target < array[middle])
		{
			end = middle - 1;
		}
		else
		{
			start = middle + 1;
		}
	}
	return -1;
}

int main()
{
	int number[] = { 0, 2, 4, 6, 8, 10, 12 };
	cout << BinarySearchIterative(number, 7, 102) << endl;
	return 0;
}